### PR TITLE
Lymon 846 guest view available experiences

### PR DESCRIPTION
**Pull request by JaaysCode**
This pull request introduces a new public API endpoint for guests to retrieve a paginated, filterable list of available experiences. It implements the required query, handler, DTOs, repository interface, and MongoDB logic, and adds comprehensive tests and fixtures to ensure correct filtering and pagination. The main changes are grouped below:

**Public API for Guest Experiences**
- Added a new `GuestExperienceController` with a `GET /guest/experiences` endpoint, allowing guests to list available experiences with pagination and optional filters for tenant, property, and category. Includes authentication and Swagger documentation.
- Registered the new controller in the `PresentationModule` to enable the endpoint. [[1]](diffhunk://#diff-176a8dc243d7c8c394be999feaf6387c82089824bd027a9e541efa457637ea80R21) [[2]](diffhunk://#diff-176a8dc243d7c8c394be999feaf6387c82089824bd027a9e541efa457637ea80R43)

**CQRS Query and Handler**
- Implemented `GetAvailableExperiencesQuery`, `GetAvailableExperiencesQueryHandler`, and `GetAvailableExperiencesResult` to handle the business logic for fetching and returning available experiences. The handler converts query parameters to value objects, applies filters, and returns paginated results. [[1]](diffhunk://#diff-454556f3481598bfb23a9277057997bb565d7f39922938d62cc5f8fc581fbef8R1-R11) [[2]](diffhunk://#diff-49d98d040faf9cbd9edf9e6671b54bdcca5a26491884dba9c35c55eb8939c91cR1-R63) [[3]](diffhunk://#diff-40dce7bc071a1eed3fcccd954d430cf5299a2c3e1e6a0539d76296bd33f7a5a5R1-R14)
- Registered the new query handler in the application module.

**Repository Layer Enhancements**
- Extended the `ExperienceRepository` interface with a new `findAvailableForGuestPaginated` method and supporting filter types, and implemented the method in the MongoDB repository. This method applies the correct filters and pagination to database queries. [[1]](diffhunk://#diff-94fa302b3c81d119003701007205dcbc5a655580ab9cb510c14087bfc9f59189R2-R15) [[2]](diffhunk://#diff-94fa302b3c81d119003701007205dcbc5a655580ab9cb510c14087bfc9f59189R32-R36) [[3]](diffhunk://#diff-da46ad7b16f261d06bc712bbebadc382e875a165d666d6831ec74a19b8391882L2-R5) [[4]](diffhunk://#diff-da46ad7b16f261d06bc712bbebadc382e875a165d666d6831ec74a19b8391882R142-R178)

**DTOs and Validation**
- Added `GetAvailableExperiencesQueryDto` for validating and transforming incoming query parameters for the new endpoint, supporting optional filters and pagination.

**Testing and Fixtures**
- Added comprehensive tests for the query handler, covering filtering, pagination, and edge cases, and created supporting fixtures and repository mocks for experiences. [[1]](diffhunk://#diff-d17e448ece20a408655615ab3177484cc903cd90edbb55c30460120e78586749R1-R173) [[2]](diffhunk://#diff-35564ead47c7daa89beffd9c024879e4d77e5e72838b670c3d4b162cdb45f5e0R1-R75) [[3]](diffhunk://#diff-ef38a39b17e2f1d144931f02823d6dfeedfcb5415b1b0953736c18329942fa5cR9)

### CODE REVIEW

File: src/application/experience/queries/GetAvailableExperiences/get-available-experiences.query-handler.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  async execute(
+    query: GetAvailableExperiencesQuery,
+  ): Promise<GetAvailableExperiencesResult> {
+    let tenantId: TenantId | undefined;
+    let propertyId: PropertyId | undefined;
+    let category: ExperienceCategory | undefined;
+
+    try {
+      if (query.tenantId) tenantId = TenantId.createFromString(query.tenantId);
+    } catch {
```
Comment (justjaaara) [inline src/application/experience/queries/GetAvailableExperiences/get-available-experiences.query-handler.ts]: No sé si es la mejor forma... Me gustaría explorar otra opción que sea más entendible...

Claude recomienda este approach pero tampoco es mi favorito:

```typescript
async execute(query: GetAvailableExperiencesQuery): Promise<GetAvailableExperiencesResult> {
  const tenantId = tryVO(() => query.tenantId ? TenantId.createFromString(query.tenantId) : undefined);
  const propertyId = tryVO(() => query.propertyId ? PropertyId.create(query.propertyId) : undefined);
  const category = tryVO(() => query.category ? ExperienceCategory.create(query.category) : undefined);

  const { experiences, total } =
    await this.experienceRepository.findAvailableForGuestPaginated(
      { tenantId, propertyId, category },
      query.page,
      query.limit,
    );

  return new GetAvailableExperiencesResult(
    experiences.map(mapExperienceToPublicDto),
    total,
    query.page,
    query.limit,
  );
}
```
```typescript
function tryVO<T>(fn: () => T | undefined): T | undefined {
  try { return fn(); } catch { return undefined; }
}
```

**Review by justjaaara — CHANGES_REQUESTED** [review]

**Review by justjaaara — DISMISSED** [review]

**Review by Lymonoficial — APPROVED** [review]
